rotkehlchen/chain/evm/decoding/balancer/v1/decoder.py

```diff
@@ -0,0 +1,392 @@
+EXIT_V1: Final = b'\xe7L\x91U+d\xc2\xe2\xe7\xbd%V9\xe0\x04\xe6\x93\xbd>\x1d\x01\xcc3\xe6V\x10\xb8j\xfc\xc1\xff\xed'  # noqa: E501
+TRANSFER_TOPIC: Final = b'\xdd\xf2R\xad\x1b\xe2\xc8\x9bi\xc2\xb0h\xfc7\x8d\xaa\x95+\xa7\xf1c\xc4\xa1\x16(\xf5ZM\xf5#\xb3\xef'  # noqa: E501
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class Balancerv1CommonDecoder(DecoderInterface, ReloadableDecoderMixin):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.pools: set[ChecksumEvmAddress] = set()
+
+    def _decode_v1_pool_event(self, all_logs: list[EvmTxReceiptLog]) -> list[dict[str, Any]] | None:  # noqa: E501
+        """Read the list of logs in search for a Balancer v1 event and return the information
+        needed to decode the transfers made in the transaction to/from the ds proxy
+        """
+        # The transfer event appears after the debt generation event, so we need to transform it
+        target_logs = [x for x in all_logs if x.topics[0] in (JOIN_V1, EXIT_V1)]
+        if len(target_logs) == 0:
+            return None
+
+        events_information = []
+        for target_log in target_logs:
+            token_address = bytes_to_address(target_log.topics[2])
+            amount = int.from_bytes(target_log.data[0:32])
+            if target_log.topics[0] == JOIN_V1:
+                balancer_event_type = BalancerV1EventTypes.JOIN
+                ds_address = bytes_to_address(target_log.topics[1])
+            else:
+                balancer_event_type = BalancerV1EventTypes.EXIT
+                ds_address = target_log.address
+
+            proxy_event_information = {
+                'ds_address': ds_address,
+                'token_address': token_address,
+                'amount': amount,
+                'type': balancer_event_type,
+            }
+            events_information.append(proxy_event_information)
+
+        return events_information
+
+    def _maybe_enrich_balancer_v1_events(
+            self,
+            context: EnricherContext,
+    ) -> TransferEnrichmentOutput:
+        """
+        Enrich balancer v1 transfer to read pool events of:
+        - Depositing in the pool
+        - Withdrawing from the pool
+
+        In balancer v1 pools are managed using a DSProxy so the account doesn't interact
+        directly with the pools.
+        """
+        events_information = self._decode_v1_pool_event(all_logs=context.all_logs)
+        if events_information is None:
+            return FAILED_ENRICHMENT_OUTPUT
+
+        for proxied_event in events_information:
+            if proxied_event['ds_address'] != context.event.address:
```

Comment:
what about key error here? Shouldn' it be handled? Or at least do a .get() to not have exceptions? Here and elsewhere
```suggestion
            if proxied_event.get('ds_address') != context.event.address:
```